### ISSUE TITLE
glfw.create_window throws uncaught exception under Wayland

### ISSUE THREAD

**Issue by troethe** [BlenderNPR/BEER#173]
See line `glfw.GLFWError: (65544) b'Wayland: Focusing a window requires user interaction'` in the attached log.
This prevents the add-on from working at all.

Also see this [issue](https://github.com/FlorianRhiem/pyGLFW/issues/42) and [this one](https://github.com/glfw/glfw/issues/1121).

[malt 2021-11-23(07-18)-censored.log](https://github.com/bnpr/Malt/files/7586104/malt.2021-11-23.07-18.-censored.log)

**Comment by troethe** [comment]
I think the solution here would be to change `glfw.ERROR_REPORTING = True` to `"warn"` or `"log"` instead.

**Comment by pragma37** [comment]
Does Malt work for you if you set glfw.ERROR_REPORTING to log?

The last time I tested Wayland there were a lot of weird errors.

**Comment by troethe** [comment]
Not quite. I have managed to get rid of all the remaining errors in the logs by setting the environment variable `PYOPENGL_PLATFORM` to `"egl"` and I think I managed to get it to render something at some point, but I sadly can't reproduce this right now. I'll maybe do a couple of more tests tomorrow.
I can understand though, if you want to wait with this until you've got confirmation that this is actually feasible on Wayland.

**Comment by pragma37** [comment]
Please keep me updated if you manage to get it working, 
I'd like to improve Linux support but sometimes can get tricky.

**Comment by troethe** [comment]
Ok, I managed to get the rendering working when pressing F12 I think: 
![render](https://user-images.githubusercontent.com/62070480/143390116-58008cce-7060-4fef-ade8-e73e759cd216.png)
On a normal gnome installation you have to set PYOPENGL_PLATFORM="egl" and it works.
However viewport rendering seems broken:
![viewport-doesnt-work](https://user-images.githubusercontent.com/62070480/143390588-b06f86e8-13c0-4b5b-a563-a376e338db69.png)

Also, when using sway, which is a tiling window manager, I have to click into the transparent Malt window every time, before it actually starts rendering. Maybe this has something to do with iconification of windows not being a feature in this window manager.

**Comment by troethe** [comment]
Nevermind, it seems like you have to `PYOPENGL_PLATFORM=osmesa blender` instead. I think this has to do with the fact, that blender still uses xwayland by default.

Anyways: On a usual Debian installation with `glfw.ERROR_REPORT="log"` and the environment variable `PYOPENGL_PLATFORM="osmesa"` I now also got the viewport rendering to work:
![viewport-works](https://user-images.githubusercontent.com/62070480/143401050-18cc7b32-ad69-493b-b51f-fc2c88eeaf6d.png)
